ouroboros-consensus-diffusion/test/consensus-test/Test/Util/TersePrinting.hs

```diff
@@ -0,0 +1,122 @@
+{-# LANGUAGE TypeFamilies #-}
+
+-- | Helpers for printing various objects in a terse way. Terse printing is
+-- similar to that provided by the 'Condense' typeclass except it can be
+-- sometimes even more compact and it is very specific to tests.
+module Test.Util.TersePrinting (
+    terseBlock
+  , terseFragment
+  , terseHFragment
+  , terseHeader
+  , tersePoint
+  , terseRealPoint
+  , terseTip
+  , terseWithOrigin
+  ) where
+
+import           Cardano.Slotting.Block (BlockNo (BlockNo))
+import           Data.List (intercalate)
+import           Data.List.NonEmpty (NonEmpty ((:|)), toList)
+import           Ouroboros.Consensus.Block (Header,
+                     Point (BlockPoint, GenesisPoint), RealPoint,
+                     SlotNo (SlotNo), blockHash, blockNo, blockSlot,
+                     realPointToPoint)
+import           Ouroboros.Network.AnchoredFragment (Anchor, AnchoredFragment,
+                     anchor, anchorToPoint, mapAnchoredFragment, toOldestFirst)
+import           Ouroboros.Network.Block (Tip (..))
+import           Ouroboros.Network.Point (WithOrigin (..))
+import           Test.Util.TestBlock (Header (TestHeader), TestBlock,
+                     TestHash (TestHash), unTestHash)
+
+-- | Group consecutive duplicate elements, counting them. Only the first element
+-- of the equality is kept. For instance:
+--
+-- > groupDuplicates [0, 0, 1, 0, 2, 2, 2] = [(2, 0), (1, 1), (1, 0), (3, 2)]
+groupDuplicates :: Eq a => [a] -> [(Int, a)]
+groupDuplicates [] = []
+groupDuplicates (z : zs) = go 1 z zs
```

Comment:
I think this function is equivalent to
```Haskell
groupDuplicates xs =  [ (length ys, head ys) | ys <- group xs ]
```

Reply:
Thank you! Used in 07c38434f183a04df600af930023afa4826d2c23.